configure.ac

```diff
@@ -1508,7 +1508,7 @@ if test -n "$RUBY"; then
         # Try Ruby1.9+ first
         RUBYDIR=`($RUBY -rrbconfig -e 'print RbConfig::CONFIG[["rubyhdrdir"]] || $rubyhdrdir') 2>/dev/null`
 	if test x"$RUBYDIR" = x"" || test x"$RUBYDIR" = x"nil"; then
-		RUBYDIR=`($RUBY -rmkmf -e 'print Config::CONFIG[["archdir"]] || $archdir') 2>/dev/null`
+		RUBYDIR=`($RUBY -rmkmf -e 'print RbConfig::CONFIG[["archdir"]] || $archdir') 2>/dev/null`
```

Comment:
On 14 January 2015 at 19:49, Klaus Kämpf <notifications@github.com> wrote:

> In configure.ac
> <https://github.com/swig/swig/pull/305#discussion-diff-22919169>:
>
> > @@ -1508,7 +1508,7 @@ if test -n "$RUBY"; then
> >          # Try Ruby1.9+ first
> >          RUBYDIR=`($RUBY -rrbconfig -e 'print RbConfig::CONFIG[["rubyhdrdir"]] || $rubyhdrdir') 2>/dev/null`
> >  	if test x"$RUBYDIR" = x"" || test x"$RUBYDIR" = x"nil"; then
> > -		RUBYDIR=`($RUBY -rmkmf -e 'print Config::CONFIG[["archdir"]] || $archdir') 2>/dev/null`
> > +		RUBYDIR=`($RUBY -rmkmf -e 'print RbConfig::CONFIG[["archdir"]] || $archdir') 2>/dev/null`
>
> Right, -rmkmf is certainly wrong. I guess it's never reached since
> RbConfig::CONFIG[["rubyhdrdir"]] is set.
>
Thanks for confirming (and that's really RbConfig::CONFIG["rubyhdrdir"] -
the extra [] are m4 quoting.

Testing with Ruby 1.9.3, these all work:

ruby -rmkmf -e 'print RbConfig::CONFIG["archdir"]'
ruby -rrbconfig -e 'print RbConfig::CONFIG["archdir"]'
ruby -e 'print RbConfig::CONFIG["archdir"]'

With 1.8.7, the first two work, but the last fails.

It's better to ask for the module we actually want rather than rely on it
being pulled in indirectly, so I've fixed this in 4fed2e6.